### PR TITLE
Fix typos in the HLLD implementation

### PR DESCRIPTION
**Pull request by AstroKriel**
A PR to fix various bugs and typos that had been introduced into the HLLD solver when being ported over to QUOKKA.

### CODE REVIEW

**Comment by markkrumholz** [comment]
@AstroKriel : once you have resolved the issues raised by the clang-tidy bot, can you mark the conversations as resolved? That way I know that you have dealt with them and the PR is ready for human review.

File: src/HLLD.hpp [hunk 1]
```diff
@@ -27,8 +27,7 @@ struct ConsHydro1D {
 	double bz;  // z-magnetic field
 };
 
-template <int N_scalars>
-AMREX_FORCE_INLINE AMREX_GPU_DEVICE auto FastMagnetoSonicSpeed(double gamma, quokka::HydroState<N_scalars> const state, const double bx) -> double
+AMREX_FORCE_INLINE AMREX_GPU_DEVICE auto FastMagnetoSonicSpeed(double gamma, quokka::MHDState const state, const double bx) -> double
```
[the remark this answers is not in the document]
Comment (AstroKriel) [reply src/HLLD.hpp:30]: Here the bot's suggestion is incorrect: the state variable is used to query the tangential magnetic field components.

File: src/HLLD.hpp [hunk 2]
```diff
@@ -40,9 +39,9 @@ AMREX_FORCE_INLINE AMREX_GPU_DEVICE auto FastMagnetoSonicSpeed(double gamma, quo
 }
 
 // HLLD solver following Miyoshi and Kusano (2005), hereafter MK5.
-template <FluxDir DIR, int N_scalars, int fluxdim>
-AMREX_FORCE_INLINE AMREX_GPU_DEVICE auto HLLD(quokka::HydroState<N_scalars> const &s_L, quokka::HydroState<N_scalars> const &s_R, const double gamma,
-					      const double bx) -> quokka::valarray<double, fluxdim>
+template <int fluxdim> // FluxDir DIR,
+AMREX_FORCE_INLINE AMREX_GPU_DEVICE auto HLLD(quokka::MHDState const &s_L, quokka::MHDState const &s_R, const double gamma, const double bx)
```
[the remark this answers is not in the document]
Comment (AstroKriel) [reply src/HLLD.hpp:43]: Again, the bot seems to aggressively assume that input arguments aren't being used. Here s_L is infact used.

**Comment by AstroKriel** [comment]
I am going to resolve all of the bot's suggestions, because they are wrong. The bot seems to suggest that the input arguments to HLLD are not being used, when they are, and also suggests that I should initialise a few variables as NAN before assigning them values: `var = NAN = 2 + 2` instead of `var = 2 + 2`.

**Comment by markkrumholz** [comment]
OK, that's fine. Please just resolve all the incorrect bot conversations and then I can review the PR.

**Comment by AstroKriel** [comment]
Done

**Review by markkrumholz — APPROVED** [review]

**Pull request merged**